dolphin/drivers/dellemc_storage/vmax/client.py

```diff
@@ -14,63 +14,115 @@
+                u4v_version=SUPPORTED_VERSION,
+                server_ip=access_info['host'],
+                port=access_info['port'],
+                verify=False,
+                array_id=array_id,
+                username=access_info['username'],
+                password=access_info['password'])
+
+        except Exception as err:
+            LOG.error("Failed to connect to VMAX: {}".format(err))
+            raise exception.StorageBackendException(
+                reason='Failed to connect to VMAX storage')
+
+    def get_version(self):
+        try:
+            # Get the VMAX version
+            return self.conn.common.get_uni_version()
+
+        except Exception as err:
+            LOG.error("Failed to get version from vmax: {}".format(err))
+            raise exception.StorageBackendException(
+                reason='Failed to get version from VMAX')
+
+    def get_model(self, symmetrix_id):
+        try:
+            # Get the VMAX model
+            uri = "/system/symmetrix/" + symmetrix_id
+            model = self.conn.common.get_request(uri, "")
+            return model['symmetrix'][0]['model']
+        except Exception as err:
+            LOG.error("Failed to get model from vmax: {}".format(err))
+            raise exception.StorageBackendException(
+                reason='Failed to get model from VMAX')
+
+    def get_storage_capacity(self, symmetrix_id):
+        try:
+            uri = "/" + SUPPORTED_VERSION + "/sloprovisioning/symmetrix/" + symmetrix_id
+            storage_info = self.conn.common.get_request(uri, "")
+            return storage_info['system_capacity']
+        except Exception as err:
+            LOG.error("Failed to get model from vmax: {}".format(err))
+            raise exception.StorageBackendException(
+                reason='Failed to get capacity from VMAX')
+
+    def list_pools(self, symmetrix_id):
+
+        try:
+            # Get list of SRP pool names
+            pools = self.conn.provisioning.get_srp_list()
+
+            pool_list = []
+            for pool in pools:
+                pool_info = self.conn.provisioning.get_srp(pool)
+
+                srp_cap = pool_info['srp_capacity']
+                total_cap = srp_cap['usable_total_tb'] * units.Ti
+                used_cap = srp_cap['usable_used_tb'] * units.Ti
+
+                status = fields.PoolStatus.AVAILABLE
+                if used_cap:
+                    status = fields.PoolStatus.IN_USE
+
+                p = {
+                    "name": pool,
+                    "storage_id": symmetrix_id,
+                    "original_id": pool_info["srpId"],
+                    "description": "Dell EMC VMAX Pool",
+                    "status": status,
+                    "storage_type": fields.StorageType.BLOCK,
+                    "total_capacity": int(total_cap),
+                    "used_capacity": int(used_cap),
+                    "free_capacity": int(total_cap - used_cap),
```

Comment:
These calculations will hold correct for SRP over Thin Data Pools? I think VMAX has all TDPs only

Reply:
These are similar to what is reported in Unisphere dashboard, virtual capacity of TDPs are not used